timemachine/fe/chiral_utils.py

```diff
@@ -55,3 +56,69 @@ def setup_chiral_bond_restraints(mol, conf, src_idx, dst_idx):
                 signs.append(-1)
 
     return np.array(restr_idxs), np.array(signs)
+
+
+def find_chiral_atoms(mol):
+    """
+    Find chiral atoms in a molecule. Note that an atom is chiral if it has an non-invertible
+    energy barrier. Even something like CH4 is considered chiral.
+
+    Parameters
+    ----------
+    mol: Chem.Mol
+        input molecule
+
+    Returns
+    -------
+    set of int
+        Chiral atoms
+
+    """
+    # these should be mutually exclusive, but if any pattern is hit then the results
+    # are accumulated to a set
+    chiral_patterns = [
+        "[X4:1]",  # any tetrahedral atom
+        "[#16X3,#15X3:1]",  # trivalent sulfur, phosphorous are assumed to be non-invertible
+        "[#7X3:1](~[R])(~[R])~[R]",  # nitrogen directly bonded to three ring atoms
+    ]
+
+    chiral_atoms = set()
+    for patt in chiral_patterns:
+        query_mol = Chem.MolFromSmarts(patt)
+        assert query_mol is not None
+        for match in mol.GetSubstructMatches(query_mol):
+            chiral_atoms.add(match[0])
+
+    return chiral_atoms
+
+
+def find_chiral_bonds(mol):
+    """
+    Find chiral bonds in a molecule:
+
+    Parameters
+    ----------
+    mol: Chem.Mol
+        input molecule
+
+    Returns
+    -------
+    set of 2-tuple
+        Chiral bonds
+
+    """
+
+    chiral_patterns = [
+        "[X2,X3:1]:[X2,X3:2]",  # any aromatic bond
+        "[X2,X3:1]=[X2,X3:2]",  # all double bonds with two or three neighbors,
+        "[NX3,NX2:1][CX3:2](=[OX1])",  # amide bond
```

Comment:
Curiosity [do not merge necessarily]: What amide bonds does the "or NX2" condition catch?

```suggestion
        "[NX3:1][CX3:2](=[OX1])",  # amide bond
```

Reply:
I might have been a bit too aggressive here, maybe something like `[H]\C(O)=N/C([H])=O`?

Reply:
Nice! Thanks for the example.

Seems reasonable to catch this one, but I don't know how edge-case-y this might be....

---------

fwiw, two observations about this molecule:

(1) rdkit doesn't draw this stereo-annotated molecule with wedge bonds,
(2) the default forcefield assigns ~2.5 + ~5 = ~7.5 kJ/mol torsion barrier to the C-N bond matched by this pattern. This is lower than the ~10 + ~3 = ~13kJ/mol barrier assigned to the C-O bond in this molecule (which is not marked as a chiral bond by this function).
<img width="422" alt="image" src="https://user-images.githubusercontent.com/5759036/173395390-1802e5e8-0b5e-4929-8c5e-2c598cde9276.png">

Reply:
1) i think wedge bond is only for tetrahedral geometries
2) the ff probably doesn't parameterize this correctly

Reply:
>i think wedge bond is only for tetrahedral geometries

Ahh, right!